functional/test/test_streams.py

```diff
@@ -145,3 +147,40 @@ def test_to_csv(self):
         sequence.to_csv(tmp_path)
         result = seq.csv(tmp_path).to_list()
         self.assertEqual(expect, result)
+
+    def test_to_sqlite3_failure(self):
+        # test failure case
+        insert_sql = "INSERT INTO user (id, name) VALUES (?, ?)"
+        elements = [(1, "Tom"), (2, "Jack"), (3, "Jane"), (4, "Stephan")]
+        with self.assertRaises(ValueError):
+            seq(elements).to_sqlite3(1, insert_sql)
+
+    def test_to_sqlite3_file(self):
+        tmp_path = 'functional/test/data/tmp/{}.db'.format(time.time())
```

Comment:
Actually I am wrong here since the DB ceases to exist once the connection is closed http://stackoverflow.com/questions/9384683/how-do-you-delete-an-in-memory-sqlite-database-with-php-pdo.

So that means what you have is correct, otherwise it wouldn't be testable